### PR TITLE
CAF-3671: Updated to remap fields  regardless of encoding

### PR DESCRIPTION
**Pull request by Anthony-McGreevy**

### CODE REVIEW

File: handlers-converters/handlers/field-mapping/src/main/java/com/github/cafdataprocessing/worker/policy/handlers/fieldmapping/FieldMappingHandler.java [hunk 1]
```diff
@@ -76,13 +80,37 @@ private void mapFields(final Map<String, String> fieldNameMappings,
...
                 .forEach(fieldNameToMap ->
                         fieldsToMap.putAll(fieldNameToMap, documentMetadata.get(fieldNameToMap)));
+    return fieldsToMap;
+    }
+
+    private Multimap<String, InputStream> getReferenceFieldsToBeMapped(final Map<String, String> fieldNameMappings,
+                                                                       final Multimap<String, InputStream> documentMetadata)
+    {
+        final Multimap<String, InputStream> fieldsToMap = ArrayListMultimap.create();
+        fieldNameMappings.keySet()
```
Comment (mmaBel) [inline handlers-converters/handlers/field-mapping/src/main/java/com/github/cafdataprocessing/worker/policy/handlers/fieldmapping/FieldMappingHandler.java]: Use CaseInsensitiveKeyMultimap here similar to metadata usage.

File: handlers-converters/handlers/field-mapping/src/test/java/com/github/cafdataprocessing/worker/policy/handlers/fieldmapping/FieldMappingHandlerTest.java [hunk 2]
```diff
@@ -150,6 +150,46 @@ public void testFieldNameSwapWithCommonTargetName() throws IOException
...
+        Assert.assertTrue(document.getMetadata().get("xyz").contains("xyz-value2"));
+
+        Assert.assertTrue(document.getMetadata().get("pqr").contains("def-value1"));
+        Assert.assertTrue(document.getMetadata().get("pqr").contains("def-value2"));
+
+        Assert.assertTrue(document.getMetadata().get("abc").isEmpty());
+        
+        Assert.assertTrue(document.getStreams().containsKey("xyz"));
+        Assert.assertTrue(document.getStreams().containsKey("pqr"));
+        Assert.assertTrue(document.getStreams().containsKey("wrx"));
```
Comment (mmaBel) [inline handlers-converters/handlers/field-mapping/src/test/java/com/github/cafdataprocessing/worker/policy/handlers/fieldmapping/FieldMappingHandlerTest.java]: Check for value here.

**Review by mmaBel — CHANGES_REQUESTED** [review]

**Comment by Anthony-McGreevy** [comment]
@mmaBel PR updated to reflect review comments.

File: handlers-converters/handlers/field-mapping/src/test/java/com/github/cafdataprocessing/worker/policy/handlers/fieldmapping/FieldMappingHandlerTest.java [hunk 3]
```diff
@@ -150,8 +151,53 @@ public void testFieldNameSwapWithCommonTargetName() throws IOException
...
+            + "  {"
+            + "    \"abc\": \"xyz\","
+            + "    \"def\": \"pqr\","
+            + "    \"pqr\": \"xyz\","
+            + "    \"jkl\": \"wrx\""
+            + "  }"
+            + "}");
+        final Long testColSeqId = 1L;
+        try (final ByteArrayInputStream stream = new ByteArrayInputStream("This is a test string".getBytes(StandardCharsets.UTF_8))) {
+            document.getStreams().put("abc", stream);
```
Comment (mmaBel) [inline handlers-converters/handlers/field-mapping/src/test/java/com/github/cafdataprocessing/worker/policy/handlers/fieldmapping/FieldMappingHandlerTest.java:172]: The try with isn't needed here. Close on a ByteArrayInput stream doesn't do anything.

https://docs.oracle.com/javase/7/docs/api/java/io/ByteArrayInputStream.html

**Review by mmaBel — APPROVED** [review]

**Pull request merged**